qa/L0_trt_dla/dla_client.py

```diff
@@ -0,0 +1,133 @@
+    resized = np.array(resized_img)
+
+    if resized.ndim == 2:
+        resized = resized[:, :, np.newaxis]
+
+    typed = resized.astype(dtype)
+
+    if scaling == 'INCEPTION':
+        scaled = (typed / 127.5) - 1
+    elif scaling == 'VGG':
+        if c == 1:
+            scaled = typed - np.asarray((128,), dtype=dtype)
+        else:
+            scaled = typed - np.asarray((123, 117, 104), dtype=dtype)
+    else:
+        scaled = typed
+
+    ordered = np.transpose(scaled, (2, 0, 1))
+
+    return ordered
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v',
+                        '--verbose',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u',
+                        '--url',
+                        type=str,
+                        required=False,
+                        default='localhost:8000',
+                        help='Inference server URL. Default is localhost:8000.')
+    parser.add_argument(
+        '-s',
+        '--scaling',
+        type=str,
+        choices=['NONE', 'INCEPTION', 'VGG'],
+        required=False,
+        default='NONE',
+        help='Type of scaling to apply to image pixels. Default is NONE.')
+    parser.add_argument('image_filename',
+                        type=str,
+                        default=None,
+                        help='Input image.')
+
+    FLAGS = parser.parse_args()
+    try:
+        triton_client = httpclient.InferenceServerClient(
+            url=FLAGS.url, verbose=FLAGS.verbose)
+    except Exception as e:
+        print("channel creation failed: " + str(e))
+        sys.exit(1)
+
+    model_name = "resnet50_plan"
+    batch_size = 32
+
+    img = Image.open(FLAGS.image_filename)
+    image_data = preprocess(img, np.int8, 3, 224, 224, FLAGS.scaling)
+    image_data = np.expand_dims(image_data, axis=0)
+
+    batched_image_data = image_data
+    for i in range(1, batch_size):
+        batched_image_data = np.concatenate((batched_image_data, image_data), axis=0)
+
+    inputs = [httpclient.InferInput('input_tensor_0', [batch_size, 3, 224, 224], 'INT8')]
+    inputs[0].set_data_from_numpy(batched_image_data, binary_data=True)
+
+    outputs = [
+        httpclient.InferRequestedOutput('topk_layer_output_index', binary_data=True)
+    ]
+
+    results = triton_client.infer(model_name,
+                        inputs,
+                        outputs=outputs)
+
+    # Validate the results by comparing with precomputed values.
+    # VULTURE class corresponds with index 23
+    EXPECTED_CLASS_INDEX = 23
+
+    output_data = results.as_numpy('topk_layer_output_index')
+    print(output_data)
+    for i in range(batch_size):
+        if output_data[i][0][0] != EXPECTED_CLASS_INDEX:
+            print("Infer error: Incorrect output label")
+            sys.exit(1)
```

Comment:
If you are already verifying the returned value in Python script, then you don't need to print and check that again in the bash script.

Reply:
Done